### PR TITLE
Refactor menu component

### PR DESCRIPTION
**Pull request by manuthecoder**
This PR greatly simplifies the codebase by converting all `menu_toggle` components to `menu__toggle` (notice the underscore!), which uses StimulusJS. 

Additionally, it improves overall user experience by preventing misclicks when closing the menu (see video for example). This is achieved by creating a transparent overlay which fills the screen. 

### Motivation
Clicking outside of menus was really annoying on the mobile app, as it would trigger buttons unintentionally.

While trying to fix this, I noticed that HCB had two menu components. The menus in the bottom navigation had a "fake" overlay by using box-shadow. 

https://github.com/hackclub/hcb/assets/77016441/cba4fecf-0273-4c14-8a65-d4976fd53040

### CODE REVIEW

File: app/assets/javascripts/ui.js [hunk 1]
```diff
@@ -295,33 +295,7 @@ $(document).on('turbo:load', function () {
...
-      $(m)
-        .siblings('[data-behavior~=menu_content]')
-        .find('input[data-behavior~=menu_input--autofocus')
-        .focus()
-    }
-    return $(m).attr('aria-expanded', !o)
-  }
-
-  if (BK.thereIs('shipping_address_inputs')) {
+    if (BK.thereIs('shipping_address_inputs')) {
```
Comment (garyhtou) [inline app/assets/javascripts/ui.js]: fix indent

**Review by garyhtou — COMMENTED** [review]
notes from call: drop overlay on desktop

**Comment by garyhtou** [comment]
I think this is good to go. @sampoder can you sanity check the edited menus and merge?

(@manuthecoder, erb/prettier should be fixed too)

**Comment by sampoder** [comment]
I'm still playing around with this but a couple of quick things:

Screenshot 2024-08-13 at 12 14 40 PM

I'm not sure about the arrow? It doesn't really fit

The amount seems to have moved for pending transactions:

Screenshot 2024-08-13 at 12 18 39 PM

**Comment by manu-gurudath** [comment]
thanks for letting me know!

I just merged `main` into this branch, so that should solve the avatar problem you're seeing

**Review by sampoder — APPROVED** [review]
Looks good to me; LGTM!

**Comment by sampoder** [comment]
@manuthecoder unfortunately merging this caused this bug: https://github.com/hackclub/hcb/issues/7590. I reverted it for now but once you get a chance to fix it, I can take another review of your PR! :D

**Pull request merged**